### PR TITLE
Add basic SQLite structure

### PR DESCRIPTION
**Pull request by connorlay**

### CODE REVIEW

File: app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package knitgrid.knitgridkotlin.presenters
+
+import android.content.Context
+
+/**
+ * Base presenter object that all presenters should descend from.
```
Comment (Magisus) [inline app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt]: Remind me again why we wanted this base presenter? Was it just to ensure they all have a context? Or did you have some additional functionality in mind for later?

File: app/src/main/kotlin/knitgrid/knitgridkotlin/data/DBQueries.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package knitgrid.knitgridkotlin.data
+
+import edu.lclark.knitgrid.data.PatternEntry
+import org.jetbrains.anko.db.classParser
+import org.jetbrains.anko.db.parseList
+import org.jetbrains.anko.db.select
+
+/**
+ * Query functions that read data from the SQLite database.
```
Comment (Magisus) [inline app/src/main/kotlin/knitgrid/knitgridkotlin/data/DBQueries.kt]: This seems straightforward enough. You said you're going to put write commands in a separate file?

File: app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt (hunk 1 above)
Comment (connorlay) [inline app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt]: I added it so I can use an extension property in `Extensions.kt` to give all presenters access to the database.

File: app/src/main/kotlin/knitgrid/knitgridkotlin/data/DBQueries.kt (hunk 2 above)
Comment (connorlay) [inline app/src/main/kotlin/knitgrid/knitgridkotlin/data/DBQueries.kt]: Yes in the `DBCommands.kt` there are more top-level functions.

File: app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt (hunk 1 above)
Comment (Magisus) [inline app/src/main/kotlin/knitgrid/knitgridkotlin/presenters/Presenter.kt]: Oh I see it now. Okay. Forgot that happens in a separate file and not in the file of the class you're altering.

**Comment by Magisus** [comment]
I like it! If you're done with it for the time being, I'll merge it and try to get it hooked into what I was doing tonight.

**Comment by connorlay** [comment]
Go ahead and merge it. I'll continue working on this stuff but I wanted you to have access to the initial stuff.

**Pull request merged**